### ISSUE TITLE
Repeated error display of "Contours are not displayed during 3D rendering"

### ISSUE THREAD

**Issue by haesleinhuepf** [napari/napari#3257]
## 🐛 Bug

Hi all, just a minor issue in 0.4.11rc2 - not hindering release of 0.4.11: The error message "Contours are not diplayed during 3D rendering" shows up again and again when `layer.data` is updated. 

## To Reproduce

1. Start napari
2. Open Cells (3D+2ch)
3. Use a labeling plugin, e.g. `Plugins > clEsperanto > Label`
4. Increase `contours` of the labels layer to 1 
5. Switch to 3D view
6. Update the labels layer, e.g. by changing sigmas in the `Label` dialog.

https://user-images.githubusercontent.com/12660498/131023056-a108abac-7f46-47b2-ac28-9d585279df10.mp4

## Expected behavior

This error should show up when the user switches to 3D or if they set `countours != 0`, not when updating layer.data.

## Environment

 - Please copy and paste the information at napari info option in help menubar here:

napari: 0.4.11rc2
Platform: Windows-10-10.0.19041-SP0
Python: 3.8.5 (default, Sep 3 2020, 21:29:08) [MSC v.1916 64 bit (AMD64)]
Qt: 5.12.9
PyQt5: 5.12.3
NumPy: 1.19.5
SciPy: 1.7.1
Dask: 2021.05.1
VisPy: 0.7.0

OpenGL:
- GL version: 4.6.0 NVIDIA 466.77
- MAX_TEXTURE_SIZE: 32768

Screens:
- screen 1: resolution 1920x1200, scale 1.0

Plugins:
- Measurements (Plot profile): 0.1.5
- Measurements (skimage regionprops): 0.1.6
- Search (Plugin): 0.1.0
- Segmentation (OCLRFC): 0.4.4
- Segmentation (split/merge): 0.1.2
- Utilities (skimage regionprops): 0.1.6
- Visualization(B/C): 0.1.1
- aicsimageio-in-memory: 0.3.5
- cellpose-napari: 0.1.3
- clEsperanto: 0.10.0
- console: 0.0.3
- napari-features: 0.1.4
- napari-yapic-prediction: 0.1.dev99
- ome-types: 0.2.7
- omero: 0.1.1
- scikit-image: 0.4.11rc2
- svg: 0.1.5

 - Any other relevant information:

**Comment by psobolewskiPhD** [comment]
I can't reproduce this in 0.4.18.
I used this script to create 3D labels.
```
import napari
from skimage import data
from scipy import ndimage as ndi

blobs = data.binary_blobs(length=128, volume_fraction=0.1, n_dim=3)
labeled = ndi.label(blobs)[0]
viewer = napari.Viewer()
viewer.add_labels(labeled, name='blob ID')
viewer.layers[0].contour = 1
viewer.dims.ndisplay = 3
```
This did result in a warning.
But then:
`viewer.layers[0].data[0:100,0:100,0:100] = 17`
and similar does not re-warn--neither does toggling 2d/3d mode for that matter.

Going to close, but if it creeps up again, please re-open or make a new issue.

**Comment by Czaki** [comment]
It was resolved by #6396
